src/finn/custom_op/fpgadataflow/hlsbackend.py

```diff
@@ -501,16 +536,28 @@ def dataoutstrm(self):
                 )
             else:
                 folded_shape = self.get_folded_output_shape(o)
-                self.code_gen_dict["$DATAOUTSTREAM$"].append(
-                    'vectorstream2npy<%s, %s, %d>(strm, %s, "%s");'
-                    % (
-                        elem_hls_type,
-                        npy_type,
-                        folded_shape[-1],
-                        oshape_cpp_str,
-                        npy_out,
+                if self.get_nodeattr("hls_style") == "freerunning":
+                    self.code_gen_dict["$DATAOUTSTREAM$"].append(
+                        'vectorstream2npy<%s, %s, %d>(strm, %s, "%s");'
+                        % (
+                            elem_hls_type,
+                            npy_type,
+                            folded_shape[-1],
```

Comment:
Just inserting
`"strm" if self.get_nodeattr("hls_style") == "freerunning" else "out0_v",`
seems to do the job of this vastly duplicating `if-else` decision.